src/components/FileDetail/FileDetailModal.js

```diff
@@ -0,0 +1,69 @@
+import React, {Fragment} from 'react';
+import UploadVersionModalContainer from './UploadVersionModalContainer';
+import AnnotateVersionModalContainer from './AnnotateVersionModalContainer';
+import NotificationBar from './NotificationBar';
+import {ReactComponent as Edit} from '../../assets/icons/edit.svg';
+
+const NotificationAction = ({onClick, buttonText}) => (
+  <button onClick={onClick} className="FileVersionList--Button">
+    <Edit className="mr-4" />
+    {buttonText}
+  </button>
+);
+
+const FileDetailModal = ({
+  className,
+  onSubmit,
+  children,
+  studyId,
+  fileNode,
+  dialog,
+  onCloseModal,
+  onNotificationClick,
+}) => {
+  return (
+    <Fragment>
+      {dialog === 'upload' && (
+        <UploadVersionModalContainer
```

Comment:
What's the point of this component if it either renders a raw `UploadVersionModalContainer` or a raw `AnnotateVersionModalContainer`?

Reply:
@dankolbman my thinking was too simple shorten the length of `FileDetail.js` by abstracting it out, but open to suggestions